### PR TITLE
Added support for the crossref REST API - api.crossref.org

### PR DESCRIPTION
**Pull request by geekscruff**
Added support for the crossref REST API - api.crossref.org

Support added for funder and journals only. There are also apis for works, members, prefixes and types.

I can't see anything on the crossref site that prevents use of the public apis in this way.

For funder:
- returns the id rather than the uri
- adds uri into the results
- constructs label and value where label is preflabel + altlabel + location (eg. National Heart, Lung, and Blood Institute, (NHLBI), United States) and value is just the preflabel (National Heart, Lung, and Blood Institute)

```
{"id":"100000050","uri":"http://dx.doi.org/10.13039/100000050","label":"National Heart, Lung, and Blood Institute, (NHLBI), United States","value":"National Heart, Lung, and Blood Institute"}
```

For journals:
- uses issn as id
- adds publisher into the results
- adds issn into results (sometimes there are two ISSNs)

```
{"id":"0730-2347","label":"Texas Heart Institute Journal","publisher":"Texas Heart Institute Journal","issn":["0730-2347","1526-6702"]},
```

Request for feedback. 
- The 'find' method returns the raw json from crossref rather than doing any processing, eg. into the same structure as the results. Is that OK?

### CODE REVIEW

**Comment by mjgiarlo** [comment]
Inclined to review this, @jcoyne @awead?

File: lib/qa/authorities/crossref/generic_authority.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def find_url(id)
+      "http://api.crossref.org/#{subauthority}/#{id}"
+    end
+
+    private
+
+      # Reformats the data received from the service
+      def parse_authority_response(response)
+        response['message']['items'].map do |result|
+          case @subauthority
```
Comment (awead) [inline lib/qa/authorities/crossref/generic_authority.rb]: shouldn't need the `@` here, I would think.

File: lib/qa/authorities/crossref/generic_authority.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    def initialize(subauthority)
+      @subauthority = subauthority
+    end
+
+    self.label = lambda do |item|
+      [item['name'],
+       item['alt-names'].blank? ? nil : "(#{item['alt-names'].join(', ')})",
+       item['location']].compact.join(', ')
+    end
```
Comment (awead) [inline lib/qa/authorities/crossref/generic_authority.rb]: This looks clever, but I'm confused by it. I don't use lambdas very much. It looks like returning an array with a single hash in it. Perhaps some YARD documentation here to explain that. Also, why this way and not just another method call?

Comment (geekscruff) [inline lib/qa/authorities/crossref/generic_authority.rb]: I based it on the existing code in geonames (and was learning how to use lambdas) ... I've added some documentation but could easily change it to a method if you think that would be more consistent / understandable?

Comment (awead) [inline lib/qa/authorities/crossref/generic_authority.rb]: @geekscruff no, that looks great... except, we use YARD docs. Sorry! Here's a reference https://gist.github.com/chetan/1827484

Comment (geekscruff) [inline lib/qa/authorities/crossref/generic_authority.rb]: Sorry, my fault. I've changed the format of the comments.

**Comment by awead** [comment]
@geekscruff thanks for sticking it out. I've approved it and this can be merged as soon as Travis is green.

**Pull request merged**